tpgtools/override.go

```diff
@@ -52,6 +52,7 @@ const (
 	CustomSerializer                   = "CUSTOM_SERIALIZER"
 	TerraformProductName               = "CUSTOM_TERRAFORM_PRODUCT_NAME"
 	UseDCLID                           = "USE_DCL_ID"
+	CustomTimeout                      = "CUSTOM_TIMEOUT"
```

Comment:
@ndmckinley @shuyama1 is this something we could obviate with a change from the DCL? I noticed the DCL sample assigns a timeout of 1800s (internal code, no easy link here, sorry) to make up for the fact that the DCL's default 15m timeout doesn't work for the resource. This should be a property of the resource so that it doesn't need to be set on a sample-by-sample basis and discoverable without an override, right?

I checked in https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/configcontroller/alpha/instance.yaml and there's no alternate timeout, at least after searching the key "timeout" "seconds" and "1800". https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/configcontroller/samples/basic_instance.yaml nor https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/configcontroller/samples/minimal_instance.yaml contain it either, so the DCL isn't passing through the information in any circumstance.

---

This part's not my call, but I'm curious if resource timeouts assigned to samples are an antipattern- we should bump the default resource timeout instead IMO.

Reply:
Agree that it could be passed through - "default timeout for apply" could, at least.  Terraform's going to need to handle timeouts for create vs update, though, since we don't have those concepts in the DCL, so the timeout override work is useful anyway.

Big agree that timeouts on a resource are the way to go, and timeouts on a sample are not - feels like a cleanup TODO to me.  @shuyama1 can you file a bug for that cleanup work?  Or if it's easy, go ahead and do it, but my impression is it's a days' work or so, and probably not the most pressing.

Reply:
> Terraform's going to need to handle timeouts for create vs update, though, since we don't have those concepts in the DCL

We can just use the same timeout for both, right? I don't think we vary it often.

Reply:
Yeah, probably - that would definitely be your call to make, and it sounds reasonable to me.  We should get that apply timeout into the openapi spec.  Maybe the delete timeout also?  @shuyama1, sorry, can you remind me, did you file a bug for that, or was I supposed to do that?

Reply:
Ah!  I remember now, you found b/188060078, which is related but not identical.  I'll comment there.

Reply:
Sorry, I will do it now and will link it here.

Reply:
No need, I found / remembered our IMs yesterday, you already did what you needed to!  :)

Reply:
Ah Okay. Just saw your comment. Thanks!